src/vm/mod.rs

```diff
@@ -174,73 +190,98 @@ fn eval_all (expressions: &[SymbolicExpression],
+                DefineResult::PersistedVariable(name, value_type, value) => {
+                    runtime_cost!(cost_functions::CREATE_VAR, global_context, value_type.size())?;
+                    contract_context.persisted_names.insert(name.clone());
+
+                    global_context.add_memory(value_type.type_size()
+                                              .expect("type size should be realizable") as u64)?;
+
+                    global_context.add_memory(value.size() as u64)?;
+
+                    global_context.database.create_variable(&contract_context.contract_identifier, &name, value_type);
+                    global_context.database.set_variable(&contract_context.contract_identifier, &name, value)?;
+                },
+                DefineResult::Map(name, key_type, value_type) => {
+                    runtime_cost!(cost_functions::CREATE_MAP, global_context,
+                                  u64::from(key_type.size()).cost_overflow_add(
+                                      u64::from(value_type.size()))?)?;
+                    contract_context.persisted_names.insert(name.clone());
+
+                    global_context.add_memory(key_type.type_size()
+                                              .expect("type size should be realizable") as u64)?;
+                    global_context.add_memory(value_type.type_size()
+                                              .expect("type size should be realizable") as u64)?;
+
+                    global_context.database.create_map(&contract_context.contract_identifier, &name, key_type, value_type);
+                },
+                DefineResult::FungibleToken(name, total_supply) => {
+                    runtime_cost!(cost_functions::CREATE_FT, global_context, 0)?;
+                    contract_context.persisted_names.insert(name.clone());
+
+                    global_context.add_memory(TypeSignature::UIntType.type_size()
+                                              .expect("type size should be realizable") as u64)?;
+
+                    global_context.database.create_fungible_token(&contract_context.contract_identifier, &name, &total_supply);
+                },
+                DefineResult::NonFungibleAsset(name, asset_type) => {
+                    runtime_cost!(cost_functions::CREATE_NFT, global_context, asset_type.size())?;
+                    contract_context.persisted_names.insert(name.clone());
+
+                    global_context.add_memory(asset_type.type_size()
+                                              .expect("type size should be realizable") as u64)?;
+
+                    global_context.database.create_non_fungible_token(&contract_context.contract_identifier, &name, &asset_type);
+                },
+                DefineResult::Trait(name, trait_type) => { 
+                    contract_context.defined_traits.insert(name, trait_type);
+                },
+                DefineResult::UseTrait(_name, _trait_identifier) => {},
```

Comment:
No -- because it's not increasing the size of the stored contract data. It _does_ get loaded when the contract-call is invoked, and that currently increases the tracked memory size, but as I mentioned in a prior comment, that is likely a check that will be eliminated in the future.

Reply:
Gotcha, thanks!